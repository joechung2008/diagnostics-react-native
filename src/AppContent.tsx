import AppTabHeaders from '@/AppTabHeaders';
import AppTabPanels from '@/AppTabPanels';
import AppToolbar from '@/AppToolbar';
import { useAppContent } from '@/hooks/useAppContent';
import { useStyles } from '@/hooks/useAppStyles';
import { ThemeProvider, useTheme } from '@/ThemeContext';
import React from 'react';
import { View } from 'react-native';

const AppContent: React.FC = () => {
  const { colors } = useTheme();
  const dynamicStyles = useStyles(colors);

  const {
    diagnostics,
    environment,
    error,
    extension,
    pending,
    selectedTab,
    setSelectedTab,
    setShowEnvironmentPicker,
    setShowList,
    showEnvironmentPicker,
    showList,
    showPaasServerless,
    handleEnvironmentChange,
    handleLinkClick,
    handlePaasServerlessPress,
    handleWebsitesPress,
  } = useAppContent();

  return (
    <ThemeProvider>
      <View style={dynamicStyles.flexbox}>
        <AppToolbar
          environment={environment}
          setShowEnvironmentPicker={setShowEnvironmentPicker}
          showEnvironmentPicker={showEnvironmentPicker}
          showPaasServerless={showPaasServerless}
          handleEnvironmentChange={handleEnvironmentChange}
          handlePaasServerlessPress={handlePaasServerlessPress}
          handleWebsitesPress={handleWebsitesPress}
        />
        <AppTabHeaders
          selectedTab={selectedTab}
          setSelectedTab={setSelectedTab}
          setShowList={setShowList}
          showList={showList}
        />
        <AppTabPanels
          diagnostics={diagnostics}
          error={error}
          extension={extension}
          pending={pending}
          selectedTab={selectedTab}
          showList={showList}
          handleLinkClick={handleLinkClick}
        />
      </View>
    </ThemeProvider>
  );
};

export default AppContent;
