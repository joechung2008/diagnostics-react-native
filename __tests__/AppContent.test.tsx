import AppContent from '@/AppContent';
import { render } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  ThemeProvider: ({ children }: { children: React.ReactNode }) => (
    <>{children}</>
  ),
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      text: '#212529',
    },
  }),
}));

// Mock hooks
jest.mock('@/hooks/useAppContent', () => ({
  useAppContent: () => ({
    diagnostics: null,
    environment: 'https://api.example.com',
    error: null,
    extension: null,
    pending: false,
    selectedTab: 'extensions',
    setSelectedTab: jest.fn(),
    setShowEnvironmentPicker: jest.fn(),
    setShowList: jest.fn(),
    showEnvironmentPicker: false,
    showList: true,
    showPaasServerless: false,
    handleEnvironmentChange: jest.fn(),
    handleLinkClick: jest.fn(),
    handlePaasServerlessPress: jest.fn(),
    handleWebsitesPress: jest.fn(),
  }),
}));

jest.mock('@/hooks/useAppStyles', () => ({
  useStyles: () => ({}),
}));

// Mock child components
jest.mock('@/AppTabHeaders', () => 'AppTabHeaders');
jest.mock('@/AppTabPanels', () => 'AppTabPanels');
jest.mock('@/AppToolbar', () => 'AppToolbar');

describe('AppContent', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<AppContent />);
    expect(toJSON()).toMatchSnapshot();
  });
});
