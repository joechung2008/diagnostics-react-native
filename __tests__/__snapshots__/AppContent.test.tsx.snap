// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AppContent renders correctly 1`] = `
<View>
  <AppToolbar
    environment="https://api.example.com"
    handleEnvironmentChange={[MockFunction]}
    handlePaasServerlessPress={[MockFunction]}
    handleWebsitesPress={[MockFunction]}
    setShowEnvironmentPicker={[MockFunction]}
    showEnvironmentPicker={false}
    showPaasServerless={false}
  />
  <AppTabHeaders
    selectedTab="extensions"
    setSelectedTab={[MockFunction]}
    setShowList={[MockFunction]}
    showList={true}
  />
  <AppTabPanels
    diagnostics={null}
    error={null}
    extension={null}
    handleLinkClick={[MockFunction]}
    pending={false}
    selectedTab="extensions"
    showList={true}
  />
</View>
`;
